contracts/feirari/FuseGuardian.sol

```diff
@@ -0,0 +1,109 @@
+      * @param cTokens The addresses of the markets (tokens) to change the supply caps for
+      * @param newSupplyCaps The new supply cap values in underlying to be set. A value of 0 corresponds to unlimited supplying.
+      */
+    function _setMarketSupplyCaps(CToken[] memory cTokens, uint[] memory newSupplyCaps) public isGovernorOrGuardianOrAdmin {
+        comptroller._setMarketSupplyCaps(cTokens, newSupplyCaps);
+    }
+
+    function _setMarketSupplyCapsByUnderlying(address[] calldata underlyings, uint[] calldata newSupplyCaps) external {
+        _setMarketSupplyCaps(_underlyingToCTokens(underlyings), newSupplyCaps);
+    }
+
+    function _underlyingToCTokens(address[] memory underlyings) internal view returns (CToken[] memory) {
+        CToken[] memory cTokens = new CToken[](underlyings.length);
+        for (uint256 i = 0; i < underlyings.length; i++) {
+            address cToken = comptroller.cTokensByUnderlying(underlyings[i]);
+            require(cToken != address(0), "cToken doesn't exist");
+            cTokens[i] = CToken(cToken);
+        }
+        return cTokens;
+    }
+
+    /**
+      * @notice Set the given borrow caps for the given cToken markets. Borrowing that brings total borrows to or above borrow cap will revert.
+      * @dev Admin or borrowCapGuardian function to set the borrow caps. A borrow cap of 0 corresponds to unlimited borrowing.
+      * @param cTokens The addresses of the markets (tokens) to change the borrow caps for
+      * @param newBorrowCaps The new borrow cap values in underlying to be set. A value of 0 corresponds to unlimited borrowing.
+      */
+    function _setMarketBorrowCaps(CToken[] memory cTokens, uint[] memory newBorrowCaps) public isGovernorOrGuardianOrAdmin {
+        comptroller._setMarketBorrowCaps(cTokens, newBorrowCaps);
+    }
+
+    function _setMarketBorrowCapsByUnderlying(address[] calldata underlyings, uint[] calldata newBorrowCaps) external {
+        _setMarketBorrowCaps(_underlyingToCTokens(underlyings), newBorrowCaps);
+    }
+
+    /**
+     * @notice Admin function to change the Borrow Cap Guardian
+     * @param newBorrowCapGuardian The address of the new Borrow Cap Guardian
+     */
+    function _setBorrowCapGuardian(address newBorrowCapGuardian) external onlyGovernor {
+        comptroller._setBorrowCapGuardian(newBorrowCapGuardian);
+    }
+
+    // ************ PAUSE GUARDIAN FUNCTIONS ************
+    /**
+     * @notice Admin function to change the Pause Guardian
+     * @param newPauseGuardian The address of the new Pause Guardian
+     * @return uint 0=success, otherwise a failure. (See enum Error for details)
+     */
+    function _setPauseGuardian(address newPauseGuardian) public onlyGovernor returns (uint) {
+        return comptroller._setPauseGuardian(newPauseGuardian);
+    }
+
+    function _setMintPausedByUnderlying(address underlying, bool state) external {
```

Comment:
I don't like how this function calls a public function that has a modifier on it and has no modifier on itself in the function signature. It obfuscates who is allowed to call it.

Prefer splitting out the _setMintPaused into an internal helper function and permissioning both functions.

Reply:
Great callout, fixed